### DISCUSSION TITLE
looking for cli for make page / layout / error / server / hook  .. etc

### DISCUSSION THREAD

**Discussion by magic-thomas** [sveltejs/kit#6607] [Q&A] [answered]
I wonder if somebody are using cli for svelte files. 

Command could be like this, I guess.

```sh
making-cmd  [type] [folder] [server/client ? ] 
```

Or can I get some similar  package for this to customize ?

**Comment by aloker** [comment]
Tbh, I don't have any need for this. I'm way faster with my IDE and a couple of snippets

**Reply by magic-thomas** [reply]
Can you explain more in detail ?

I am using VSCodium( VSCode) .

**Reply by jycouet** [reply]
I'm also going Alllll-in with snippets.

For example, I have: 
```json
"load": {
    "prefix": "JYC2_Load",
    "body": [
      "import type { PageLoad } from './\\$types';",
      "",
      "export const load: PageLoad = async (event) => {",
      "  $1",
      "  return {};",
      "};"
    ],
    "description": "load"
  },
```

And 💥, I have my load function, and typed!

2 small tricks to notice: 
- I have my snippets always starting with `JYC` so I know how to find them easily.
- I put a number to order them in _**my**_ convenient way


![image](https://user-images.githubusercontent.com/5312607/188708967-de4e5999-4f99-4546-b1de-3ba633284cea.png)

**Reply by magic-thomas** [reply]
Thanks for kind rely. 


* +error.svelte   
* +layout.svelte  
* +page.server.js 
* +server.js
* +layout.js      
* +page.js        
* +page.svelte

I made 7 snippets for all cases.

Actually I really like Laravel cli package. It make coding funny.

So I just wish I can make cli for all cases to see all type list and the contents at a glance.

**Comment by teemingc** [comment]
You may be interested in https://sveltin.io/
Although, I saw this a long time ago, and am not sure if it's actively maintained.

**Reply by magic-thomas** [reply]
oh .. looks good and  maintained recently. 

I will check it. 

Thanks bro.

**Reply by magic-thomas** [reply]
I check this for a while. 

It looks too heavy to handle for new user. 

It create event svelte app with their own command, that might not be desirable for common engineers. 

Thanks for information.

**Comment by j4w8n** [comment, marked as answer]
I created this after the recent routing changes.

https://github.com/j4w8n/mkrt

**Reply by magic-thomas** [reply]
Thanks for reply. 

It looks handy and easily expandable. 

I will try. 

Thanks.

**Comment by jycouet** [comment]
👋 again @magic-thomas 

I put together my favorite solution, let me share 🤟
It's snippets... at another level!

1/ Add the Extension `Folder Template`
Description: Create Templates for Folders and Files for easy reusability
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=Huuums.vscode-fast-folder-structure`

2/ Create a folder in the root of your project `.fttemplates`
Then, each subfolder will be a template... You can have multiple files... you can also have variables replacement! It's crazy good.

3/ That's it. Just use it 🥳

---

I just created an example here: https://github.com/jycouet/kit-folder-template

What is great also is that it's linked to the project... so your colleagues can use them, tune them, commit them 🥳

**Reply by magic-thomas** [reply]
Thanks for sharing your solution. 

That is an another impressive solution and looks interesting too. 

I feel happy, I can choose anything among 'bests'. 

Thanks.
